stor/dx.py

```diff
@@ -144,41 +138,273 @@ def download_objects(self):  # may not need it
         raise NotImplementedError
 
     def remove(self):
-        raise NotImplementedError
+        """Removes a single object from DX platform
+
+        Raises:
+            ValueError: The path is invalid.
+        """
+        if not self.resource:
+            raise ValueError('DXPath must point to single data object to call remove')
+        file_handler = dxpy.DXFile(dxid=self.canonical_resource,
+                                   project=self.canonical_project)
+        file_handler.remove()
 
     def rmtree(self):
-        raise NotImplementedError
+        """
+        Removes a resource and all of its contents. The path should point to a project or directory.
+
+        Raises:
+            ValueError: The path points to a file
+        """
+        proj_handler = dxpy.DXProject(self.canonical_project)
+        if not self.resource:
+            return proj_handler.destroy()
+        try:
+            proj_handler.remove_folder(self.resource, recurse=True)
+        except dxpy.exceptions.ResourceNotFound as e:
+            raise NotFoundError('No folders were found with the given path ({})'
+                                .format(self), e)
+
+    def makedirs_p(self):
+        """Make directories from DX folder paths.
+
+        The resulting folders will have same access permissions as the project.
```

Comment:
This line is here because makedirs_p in other places takes in a mode argument (default 0o777) which describes the permission towards the newly created directories. Here, we don't have control over folder level permissions since they are decided at the project level.
And makedirs_p makes directories for a single path, doesn't take a set of folder paths in. I modified the docstring to merge yours and mine.

Reply:
again take this line out, can move it into mode line:
```
Args:
    mode: unused, present for compatibility (access permissions are managed at project level)
```

Reply:
Done